### DISCUSSION TITLE
How to wait after closing window before heat is turned on?

### DISCUSSION THREAD

**Discussion by jtulak** [jmcollin78/versatile_thermostat#724] [Q&A] [answered]
Hi. As far as I understand the [window feature](https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/feature-window.md), _window sensor delay_ is used for both opening and closing detection. But I would like to have two vastly different values:
- when a window opens, give it 30 sec and then stop
- when a window closes, wait 10 minutes for temperature to stabilize rather than crank up the heat just because the cold air was blowing over the thermometer.

Is there some configuration I'm overlooking? I know I can use a template sensor with delay_on/delay_off options to make the window appear open after it was closed, but that's just a workaround.

**Comment by jmcollin78** [comment, marked as answer]
Hello, this have been claim many times but this is useless. Just let the VTherm turn on and it will adapts it cycle quickly when temperature goes up.

**Reply by joshua-lehmann** [reply]
Hi @jmcollin78 
Hi Jean-Marc
Thank you for your fantastic integration! I’m requesting the same feature as the original poster and would like to respectfully disagree with your opinion on its necessity.

I have centralized oil heating, and my apartment's billing is based on sensors attached to each radiator. These sensors calculate usage by measuring the radiator's temperature relative to the room temperature. When I open a window—especially in the current cold weather (around 0°C)—the radiator heats up rapidly to compensate, even though the room temperature takes much longer to recover. Most of my radiators, and therefore the cost-calculating sensors, are directly below windows it leads to higher costs because the sensors detect increased radiator temperatures immediately after closing a window, even though the room temperature remains low.

If there were a feature to delay heating for 30 minutes after a window is closed, it could save significant energy and costs. By waiting, the room temperature would naturally rise closer to the desired level, reducing the need for the radiator to work at full power. This would be a highly useful feature in my setup.

Thank you for considering this!

**Comment by jmcollin78** [comment]
Hello @joshua-lehmann ,

Thank you for this explaination. I was looking for a real case explained like this.

I'm not a expert but in my opinion (I can be wrong):
1. the most you wait, the most the heating energy on the room is lost and the most you will have to heat to recover the same level,
2. moreover, your heater will be colder and will take time to rise up again (you have to heat the heater). This is specially true if your heating system have a long inertia,
3. best heating mecanism should be the most stable possible and avoid down and up heating cycle. If you wait let's says 30 min, this will generate a big fall which is (in my understanding) not desirable,
4. when you let "the room temperature naturally rise closer", it will take the heating energy for elsewhere (other rooms, equipments, ...) and you have to re-heat all of this other rooms equipment, etc.

So I'm really not sure (but once again I'm not an expert and more a developer) this is a good idea in a economic point of view.

After all this is not a big challenge to add one more parameter. The only point is that the configuration is already complex (more than 60 parameters and combinations to deal with) and if it becomes too complex, the integration will die from over-complexity (and nobody wants this). So I'm very vigilant of this.

**Reply by joshua-lehmann** [reply]
Hi @jmcollin78  

Thank you for your detailed response and for considering the feature request! I’d like to share some additional thoughts based on my setup:  

1. **Radiator Behavior and Costs**:  
   - My radiators don’t cool down very quickly, so there’s minimal energy loss from waiting.  
   - When I open all windows, the room temperature can drop from 20°C to 14°C very quickly. This creates a large difference between the radiator (which remains hot) and the room, leading to high “usage” being recorded by the sensors—even if the valve is closed. Unfortunately, this outdated measurement method is still common in many apartments.  
   - If I wait ~20 minutes after closing the windows, the room naturally warms back to ~18°C due to insulation and residual heat from other rooms, saving both energy and costs by reducing unnecessary radiator activity.  

2. **UI Complexity**:  
   I understand the concern about configuration complexity, but perhaps a wizard could guide users through the setup process and explain key options. I also noticed that the question mark links in the configuration window currently redirect to the main GitHub page rather than specific documentation for the option in question. This could be improved to provide users with more targeted guidance.  

3. **Documentation and Understanding**:  
   While I’ve managed to set things up, I struggled with some terms like *EMA temp*, *slope*, and *off/on time*:  
   - **EMA temp**: I understand it relates to the exponentially weighted moving average, but it wasn’t clear how its used or adjusted.  
   - **Slope**: While I assume it’s used to fine-tune how the temperature is calculated or controlled, I’d appreciate a clearer explanation of its function and practical examples.  
   - **Off/on time with direct valve control**: This part was particularly confusing. If it’s direct control, is it not always “on” but rather at a percentage?  
   - Regulation Threshold: In Over Climate with Direct Valve Controll, is it in degrees or % for valve? Since this setup is a mix of valve/over climate I was not really able to figure it out.
   A step-by-step example configuration or practical guide for these terms would be incredibly helpful, especially for new users.  

I’m a developer myself and would be happy to help improve the documentation or contribute to a potential wizard to make the integration easier to use. Let me know if there’s anything I can do to assist! But at the moment I am still learning and do not understand everything well enough.

Thanks again for your hard work and for listening to user feedback—I truly appreciate the versatility and potential of this integration!  

Best regards,  
Joshua

**Comment by jmcollin78** [comment]
Hello @joshua-lehmann ,

So many thanks for this long report and your proposals to help. This is a great news and I will consider this proposal for collaboration.

> contribute to a potential wizard to make the integration easier to use

The HA default configuration UI mecanism (ConfigFlow) is very rigid and we cannot do what we want. So I guess, this is not a good idea.

My though was to do some videos to explain how it works and how to configure the VTherm. I think this is certainly the best format. Documentation is boring and many people just don't read or doesn't understand. I have so many examples, even in the French forum (so it is not a not native language barrier). Me the first, when I read a long documentation not in my native language, I have difficulties to understand all the concepts

But I have some strong limitations:
1. I have no strong skills to make this. I have to learn all aspects for this,
2. My english is too bad. Ideally, videos should in English with maybe subtitles in differents languages. I'm not able to do this in English directly,
3. I'm convinced that it is a hard work and time consuming. And I have no time for a big project like this for now.

> Let me know if there’s anything I can do to assist! But at the moment I am still learning and do not understand everything well enough.

I don't know if it something you can help, if I decide to start something like that ?

Maybe we can continue this discussion elsewhere.

Let's focus on the close window delay on this thread. I will certainly reconsider my position because it is requested by many users.

**Reply by joshua-lehmann** [reply]
HI @jmcollin78 
Thanks for your detailed answer, I am happy to continue the discussion about documentation YT video etc. somewhere else, feel free to contact me via the contact details on my GitHub Profile (since GitHub does not have any private messages)

**Comment by jmcollin78** [comment]
I reopen this feature request: https://github.com/jmcollin78/versatile_thermostat/issues/791

**Comment by jmcollin78** [comment]
https://github.com/jmcollin78/versatile_thermostat/releases/tag/7.1.2
